src/Microsoft.Data/DataFrame.cs

```diff
@@ -33,6 +36,273 @@ public DataFrame(IList<BaseColumn> columns)
             _table = new DataFrameTable(columns);
         }
 
+        public DataFrame(RecordBatch recordBatch)
+        {
+            _table = new DataFrameTable();
+            Apache.Arrow.Schema arrowSchema = recordBatch.Schema;
+            int fieldIndex = 0;
+            IEnumerable<IArrowArray> arrowArrays = recordBatch.Arrays;
+            foreach (IArrowArray arrowArray in arrowArrays)
+            {
+                Field field = arrowSchema.GetFieldByIndex(fieldIndex);
+                IArrowType fieldType = field.DataType;
+                BaseColumn dataFrameColumn = null;
+                Memory<byte> dataFrameMemory, nullBitMapMemory;
+                bool copied;
+                switch (fieldType.TypeId)
+                {
+                    case ArrowTypeId.Boolean:
+                        ReadOnlyMemory<byte> valueBuffer = (arrowArray as BooleanArray).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> nullBitMapBuffer = (arrowArray as BooleanArray).NullBitmapBuffer.Memory;
+                        dataFrameMemory = new Memory<byte>(new byte[valueBuffer.Length]);
+                        nullBitMapMemory = new Memory<byte>(new byte[nullBitMapBuffer.Length]);
+                        copied = valueBuffer.TryCopyTo(dataFrameMemory) && nullBitMapBuffer.TryCopyTo(nullBitMapMemory);
```

Comment:
I would have hoped we could do this without a copy....

Reply:
Yup, I do plan to support that soon(https://github.com/dotnet/corefxlab/issues/2697). The problem is that Arrow exposes only a ReadOnlyMemory<byte>